cmd/docker/docker.go

```diff
@@ -137,7 +137,7 @@ func setHelpFunc(dockerCli command.Cli, cmd *cobra.Command) {
 			return
 		}
 
-		if len(args) >= 1 {
+		if len(args) >= 1 && ccmd.Root() == ccmd.Parent() {
```

Comment:
Thank you for working on this!!

I'm slightly trying to get my head around the fix; and probably it could need a comment in the code, describing what we're doing here, and a similar comment in the commit message.

Also (not yet fully grasping the fix 😓 , so perhaps silly question); this only takes `.Parent()` into account, does that affect cases where "nested" commands could run into this?

Reply:
Hey! 👋🏻  I enjoyed working on this one, even tho it's an easy fix I struggled to make `dlv` debugger to work in vscode and then to find the issue, but it was fun 

So yeah I know, I need to add comments and maybe do it in a better way like create a function that does the check and returns a boolean

The idea is to identify if the command is first level command, in the case of `docker swarm init --help`
```
docker: is the root command
swarm : 1st level command
init  : 2nd level command
--help: is a flag
```

Why to idenfity 1st only? because docker plugins are in the form `docker-<plugin name>`, for this reason it was taking 2nd command `init` and making it `docker-init`

So how do we identify if it's the 1st level? checking parent command and root command, if both are the same then my command it's 1st level command - in the `init` case, root cmd is `docker` and parent is `swarm` so it will fail, no need to check great parent or nested

Yeah definitely will improve the commit, this was my first shot - thanks for the review!

Reply:
(quick comment, in a call right now)

> and maybe do it in a better way like create a function that does the check and returns a boolean

I'm fine with keeping the code in-line. Abstracting things can also hide logic, and sometimes it's fine to just have the logic in plain sight (not "shoved under the rug" to hide it away)

Reply:
added comments and rewrote commit msg, any help with wording will be highly appreciated 🙏🏻

Reply:
Hello @thaJeztah changes has been performed, can this be reviewed again? 🙏🏻